src/webview-src/components/App.tsx

```
@@ -33,37 +53,194 @@ function StatusDot({ status }: { status: 'online' | 'offline' | 'connecting' })
       </div>
-      {text ? (
+      <div className="font-semibold mt-2">Rejection Reason:</div>
+      <div className="whitespace-pre-wrap mt-1">{event.rejection_reason}</div>
+    </div>
+  );
+}
+
+function AgentErrorBlock({ event }: { event: AgentErrorEvent }) {
+  return (
+    <div className="bg-[rgba(220,0,0,0.08)] border-l-[3px] border-[rgba(220,0,0,0.7)] p-3 rounded my-1">
+      <div className="font-bold mb-2 text-red-600">Agent Error</div>
+      <div className="font-semibold">Error Details:</div>
+      <div className="whitespace-pre-wrap mt-1 text-red-700">{event.error}</div>
+      {event.tool_name && (
+        <div className="mt-2 text-sm opacity-70">Tool: {event.tool_name}</div>
+      )}
+    </div>
+  );
+}
+
+function PauseEventBlock({ event }: { event: PauseEvent }) {
+  return (
+    <div className="bg-[rgba(255,200,0,0.1)] border-l-[3px] border-[rgba(255,200,0,0.8)] p-3 rounded my-1">
+      <div className="font-bold text-yellow-700">User Paused</div>
+      <div className="mt-1 text-sm opacity-80">Conversation Paused</div>
+    </div>
+  );
+}
+
+function CondensationBlock({ event }: { event: Condensation }) {
+  return (
+    <div className="bg-[rgba(200,50,200,0.06)] border-l-[3px] border-[rgba(200,50,200,0.6)] p-3 rounded my-1">
+      <div className="font-bold mb-2">Auto Conversation Condensation Triggered</div>
+      <div>Forgetting {event.forgotten_event_ids.length} events</div>
+      {event.summary && (
+        <>
+          <div className="font-semibold mt-2">[Summary of Events Being Forgotten]</div>
+          <div className="whitespace-pre-wrap mt-1">{event.summary}</div>
+        </>
+      )}
+    </div>
+  );
+}
+
+function MessageEventBlock({ event }: { event: AgentMessageEvent }) {
+  const role = event.llm_message.role;
+  const parts = event.llm_message.content.filter(isTextContent).map((c) => c.text);
+  const content = parts.join('\n');
```

:frame_photo: Feature drop: Non-text content (images) is silently ignored. Given ImageContent now uses image_urls[], render a simple gallery or at least a placeholder.